### PR TITLE
Add markdown to apps forms descriptions and errors

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add markdown to apps forms field descriptions and errors.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35179

#### Related PRs
Webapp: https://github.com/mattermost/mattermost-webapp/pull/8219

#### Screenshots
Screenshot from 2021-06-04 17-28-51
Screenshot from 2021-06-04 17-29-02
Screenshot from 2021-06-04 17-29-11
Screenshot from 2021-06-04 17-29-22
Screenshot from 2021-06-07 16-01-07
Screenshot from 2021-06-07 16-01-32

#### Release Note
```release-note
Add markdown support for app fields descriptions and errors.
```

### CODE REVIEW

**Review by catalintomai — APPROVED** [review]

**Comment by larkox** [comment]
I also added markdown for command descriptions. I added some screenshots on how it looks.

**Review by catalintomai — APPROVED** [review]

File: app/components/autocomplete/slash_suggestion/slash_suggestion_item.tsx [hunk 1]
```diff
@@ -130,13 +134,12 @@ const SlashSuggestionItem = (props: Props) => {
...
-                        numberOfLines={1}
-                        style={style.suggestionDescription}
-                    >
-                        {description}
-                    </Text>
+                    <Markdown
+                        baseTextStyle={style.suggestionDescription}
+                        textStyles={getMarkdownTextStyles(theme)}
+                        blockStyles={getMarkdownBlockStyles(theme)}
+                        value={description}
```
Comment (enahum) [inline app/components/autocomplete/slash_suggestion/slash_suggestion_item.tsx:141]: do we want to set any of these as true?

* disableHashtags
* disableAtMentions
* disableChannelLink
* disableAtChannelMentionHighlight
* disableGallery

File: app/components/autocomplete_selector/autocomplete_selector.js [hunk 2]
```diff
@@ -167,18 +171,28 @@ export default class AutocompleteSelector extends PureComponent {
...
             helpTextContent = (
-                <Text style={style.helpText}>
-                    {helpText}
-                </Text>
+                <View style={style.helpTextContainer} >
+                    <Markdown
+                        baseTextStyle={style.helpText}
+                        textStyles={textStyles}
+                        blockStyles={blockStyles}
+                        value={helpText}
```
Comment (enahum) [inline app/components/autocomplete_selector/autocomplete_selector.js:179]: same as above

**Review by enahum — CHANGES_REQUESTED** [review]
Please consider disabling the features from Markdown that would have a negative impact with this PR

File: app/components/autocomplete/slash_suggestion/slash_suggestion_item.tsx (hunk 1 above)
Comment (larkox) [reply app/components/autocomplete/slash_suggestion/slash_suggestion_item.tsx:141]: After this conversation [here](https://community.mattermost.com/core/pl/tiapzoetg7b33dx8yjji3xkbdo) we will probably remove the markdown support completely from slash commands.

File: app/components/autocomplete_selector/autocomplete_selector.js (hunk 2 above)
Comment (larkox) [reply app/components/autocomplete_selector/autocomplete_selector.js:179]: Highlights and hashtags I think we should disable. Channel and User mentions... I see value, but I am not 100% sure. Gallery the same. I see value the moment we have the image metadata passed through.

Comment (enahum) [reply app/components/autocomplete_selector/autocomplete_selector.js:179]: maybe give it a try and see what is the UX when you tap on them

**Review by enahum — APPROVED** [review]

**Comment by DHaussermann** [comment]
@larkox This is looking good. But, I have couple questions...

1. I don't see the extra text and markdown in the `/test` slash command should this work in the `master` branch of the test-app
2. In the modal checked lists show as the raw text. Is this expected? The same happens when just making a post in a channel. Maybe this is just how the RN markdown parser works?
image
image

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Markdown in descriptions and errors is displaying as expected
- Tested on Android and iOS
- Remaining 2 questions have been addressed 
  - Checkboxes always show this way on RN
  - Rendering markdown in the slash command was cut from the scope. Only raw text will be shown.
- No issues found.
  
 LGTM!

**Pull request merged**